generatebundlefile/promote.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+)
+
+type SDKClients struct {
+	ecrPublicClient        *ecrPublicClient
+	stsClient              *stsClient
+	ecrClient              *ecrClient
+	ecrPublicClientRelease *ecrPublicClient
+}
+
+func NewAWSSessionProfile(profile, region string) (aws.Config, error) {
+	if profile == "" || region == "" {
+		return aws.Config{}, fmt.Errorf("Empty profile or region passed to function")
+	}
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithRegion(region),
+		config.WithSharedConfigProfile(profile),
+	)
+	if err != nil {
+		return aws.Config{}, fmt.Errorf("Creating session with profile")
+	}
+	return cfg, nil
+}
+
+// GetSDKClients is used to handle the creation of different SDK clients.
+func GetSDKClients(profile string) (*SDKClients, error) {
+	clients := &SDKClients{}
+	var err error
+	clients.ecrPublicClient, err = NewECRPublicClient(true, nil)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create SDK connection to ECR Public %s", err)
+	}
+	clients.stsClient, err = NewStsClient(true)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create SDK connection to STS %s", err)
+	}
+	clients.ecrClient, err = NewECRClient(true)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create SDK connection to ECR %s", err)
+	}
+	if profile != "" {
+		cfg, err := NewAWSSessionProfile(profile, ecrPublicRegion)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to create SDK connection to session to another profile %s", err)
+		}
+		clients.ecrPublicClientRelease, err = NewECRPublicClient(true, &cfg)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to create SDK connection to ECR Public another profile %s", err)
+		}
+	}
+	return clients, nil
+}
+
+// PromoteHelmChart will take a given repository, and authFile and handle helm and image promotion for the mentioned chart.
+func (c *SDKClients) PromoteHelmChart(repository, authFile string, profile bool) error {
```

Comment:
Should `profile` be named `public`?

Reply:
It's about whether or not it's using a profile to perform cross account promotion, I will change to be more clear.